diff --git git/src/drivers/driver_nl80211.c git/src/drivers/driver_nl80211.c
index fce6efd..8f52df5 100644
--- git/src/drivers/driver_nl80211.c
+++ git/src/drivers/driver_nl80211.c
@@ -4340,28 +4340,6 @@ out_err:
 }
 
 
-static int nl80211_mgmt_subscribe_ap_dev_sme(struct i802_bss *bss)
-{
-	if (nl80211_alloc_mgmt_handle(bss))
-		return -1;
-	wpa_printf(MSG_DEBUG, "nl80211: Subscribe to mgmt frames with AP "
-		   "handle %p (device SME)", bss->nl_mgmt);
-
-	if (nl80211_register_frame(bss, bss->nl_mgmt,
-				   (WLAN_FC_TYPE_MGMT << 2) |
-				   (WLAN_FC_STYPE_ACTION << 4),
-				   NULL, 0) < 0)
-		goto out_err;
-
-	nl80211_mgmt_handle_register_eloop(bss);
-	return 0;
-
-out_err:
-	nl_destroy_handles(&bss->nl_mgmt);
-	return -1;
-}
-
-
 static void nl80211_mgmt_unsubscribe(struct i802_bss *bss, const char *reason)
 {
 	if (bss->nl_mgmt == NULL)
@@ -7866,10 +7844,6 @@ static int nl80211_setup_ap(struct i802_bss *bss)
 		if (nl80211_mgmt_subscribe_ap(bss))
 			return -1;
 
-	if (drv->device_ap_sme && !drv->use_monitor)
-		if (nl80211_mgmt_subscribe_ap_dev_sme(bss))
-			return -1;
-
 	if (!drv->device_ap_sme && drv->use_monitor &&
 	    nl80211_create_monitor_interface(drv) &&
 	    !drv->device_ap_sme)
-- 
1.8.1.4

